src/KeyDB.php

```diff
@@ -73,37 +74,41 @@ public static function getSupportedKeyTypes(): array
     }
 
     /**
-     * Computes the hashed keys IDs for the given key metadata.
+     * Computes the key ID for the given key metadata.
+     *
+     * Per specification section 4.2, the KEYID is a hexdigest of the SHA-256
+     * hash of the canonical form of the key.
      *
      * @param \ArrayAccess $keyMeta
      *     An ArrayAccess object of key metadata. See self::addKey() and the TUF
      *     specification for the array structure.
      *
-     * @return string[]
-     *     An array of hashed key IDs for the key metadata. Each entry is a
-     *     string hash of the key signature and all its associated metadata.
-     *     There is one entry for each hashing algorithm specified in the
-     *     'keyid_hash_algorithms' child array.
+     * @return string
+     *     The key ID in hex format for the key metadata hashed using sha256.
      *
      * @see https://github.com/theupdateframework/specification/blob/v1.0.9/tuf-spec.md#4-document-formats
      *
      * @todo https://github.com/php-tuf/php-tuf/issues/56
      */
-    private static function computeKeyIds(\ArrayAccess $keyMeta): array
+    private static function computeKeyId(\ArrayAccess $keyMeta): string
     {
+        // @see https://github.com/secure-systems-lab/securesystemslib/blob/master/securesystemslib/keys.py
+        // The keyid_hash_algorithms array value is based on the TUF settings,
+        // it's not expected to be part of the key metadata. The fact that it is
+        // currently included is a quirk of the TUF python code that may be
+        // fixed in future versions. Calculate using the normal TUF settings
+        // since this is how it's calculated in the securesystemslib code and
+        // any value for keyid_hash_algorithms in the key data in root.json is
+        // ignored.
         $keyCanonicalStruct = [
             'keytype' => $keyMeta['keytype'],
             'scheme' => $keyMeta['scheme'],
-            'keyid_hash_algorithms' => $keyMeta['keyid_hash_algorithms'],
+            'keyid_hash_algorithms' => ['sha256', 'sha512'],
             'keyval' => ['public' => $keyMeta['keyval']['public']],
         ];
         $keyCanonicalForm = JsonNormalizer::asNormalizedJson($keyCanonicalStruct);
 
-        // Generate a hash of the key and its metadata for each of the listed
-        // keyid_hash_algorithms.
-        return array_map(function ($algo) use ($keyCanonicalForm) {
-            return hash($algo, $keyCanonicalForm, false);
-        }, $keyMeta['keyid_hash_algorithms']);
+        return hash('sha256', $keyCanonicalForm, false);
```

Comment:
Is this the default algo setting that securesystemslib or the python TUF client will use or will this always be `sha256` when creating the server json files?

Reply:
The TUF client always uses just the sha256 version. That's what's in the spec.

Reply:
ok. thanks I found this in the spec

> The KEYID of a key is the hexdigest of the SHA-256 hash of the canonical form of the key.